tools/rosconsole/src/rosconsole/rosconsole.cpp

```diff
@@ -393,6 +396,13 @@ void Formatter::print(void* logger_handle, ::ros::console::Level level, const ch
   ss << COLOR_NORMAL;
 
   fprintf(f, "%s\n", ss.str().c_str());
+  
+  if (g_force_stdout_line_buffered && f == stdout) {
+    int flushResult = fflush(f);
+    if (flushResult != 0) {
+      fprintf(stderr, "Failed to perform fflush on stdout, return code is %d\n", flushResult);
```

Comment:
If the flushing fails I don't think the repeated error message for every case is helpful. I would suggest to only print the error message for the first occurrence.